clickhouse-jdbc/src/main/java/com/clickhouse/jdbc/ConnectionImpl.java

```diff
@@ -31,6 +36,46 @@ public ConnectionImpl(String url, Properties info) {
                 .build();
     }
 
+    public String getUser() {
+        return config.getUser();
+    }
+
+    public String getURL() {
+        return url;
+    }
+
+    private String getServerVersion() throws SQLException {
+        try (QueryResponse response = client.query("SELECT version()").get(30, TimeUnit.SECONDS)) {
+            // Create a reader to access the data in a convenient way
+            ClickHouseBinaryFormatReader reader = client.newBinaryFormatReader(response);
```

Comment:
I would avoid creating a reader for small datasets - I usually use `com.clickhouse.data.ClickHouseFormat#TSKV` format because it can be read `java.util.Properties` - see reference code: `com.clickhouse.client.api.internal.TableSchemaParser#readTSKV`

Reply:
or `queryAll` API that returns generic records list.